app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,294 @@
+        addPreferencesFromResource(R.xml.preferences_synchronization);
+        setupScreen();
+        updateScreen();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setTitle(R.string.synchronization);
+        updateScreen();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        EventBus.getDefault().unregister(this);
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setSubtitle("");
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN, sticky = true)
+    public void syncStatusChanged(SyncServiceEvent event) {
+        updateScreen();
+        if (!GpodnetPreferences.loggedIn()) {
+            return;
+        }
+        if (event.getMessageResId() == R.string.sync_status_error
+                || event.getMessageResId() == R.string.sync_status_success) {
+            updateLastSyncReport(SyncService.isLastSyncSuccessful(getContext()),
+                    SyncService.getLastSyncAttempt(getContext()));
+        } else {
+            ((PreferenceActivity) getActivity()).getSupportActionBar().setSubtitle(event.getMessageResId());
+        }
+    }
+
+    private void setupScreen() {
+        final Activity activity = getActivity();
+
+        findPreference(PREFERENCE_LOGIN).setOnPreferenceClickListener(preference -> {
+            AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            builder.setTitle(R.string.dialog_choose_sync_service_title);
+            String[] syncProviderDescriptions = {
+                    getString(R.string.gpodnet_description),
+                    getString(R.string.preference_synchronization_summary_nextcloud)
+            };
+            int[] syncProviderIcons = {R.drawable.gpodder_icon, R.drawable.nextcloud_logo_svg};
+
+            ListAdapter adapter = new ArrayAdapter<String>(
+                    getContext(), R.layout.alertdialog_sync_provider_chooser, syncProviderDescriptions) {
+
+                ViewHolder holder;
+
+                class ViewHolder {
+                    ImageView icon;
+                    TextView title;
+                }
+
+                public View getView(int position, View convertView, ViewGroup parent) {
+                    final LayoutInflater inflater = LayoutInflater.from(getContext());
+                    if (convertView == null) {
+                        convertView = inflater.inflate(
+                                R.layout.alertdialog_sync_provider_chooser, null);
+
+                        holder = new ViewHolder();
+                        holder.icon = (ImageView) convertView
+                                .findViewById(R.id.icon);
+                        holder.title = (TextView) convertView
+                                .findViewById(R.id.title);
+                        convertView.setTag(holder);
```

Comment:
This is taking a lot of vertical space. Please break lines only when they reach 120 characters.